lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -85,23 +89,41 @@ struct CompRegLower : public OpConversionPattern<OpTy> {
     circt::sv::setSVAttributes(svReg, circt::sv::getSVAttributes(reg));
 
     auto regVal = rewriter.create<sv::ReadInOutOp>(loc, svReg);
+
+    auto assignValue = [&] { createAssign(rewriter, svReg, reg); };
+    auto assignReset = [&] {
+      rewriter.create<sv::PAssignOp>(loc, svReg, reg.getResetValue());
+    };
+
     if (reg.getReset() && reg.getResetValue()) {
-      rewriter.create<sv::AlwaysFFOp>(
-          loc, sv::EventControl::AtPosEdge, reg.getClk(), ResetType::SyncReset,
-          sv::EventControl::AtPosEdge, reg.getReset(),
-          [&]() { createAssign(rewriter, svReg, reg); },
-          [&]() {
-            rewriter.create<sv::PAssignOp>(loc, svReg, reg.getResetValue());
-          });
+      if (lowerToAlwaysFF) {
+        rewriter.create<sv::AlwaysFFOp>(
+            loc, sv::EventControl::AtPosEdge, reg.getClk(),
+            ResetType::SyncReset, sv::EventControl::AtPosEdge, reg.getReset(),
```

Comment:
This used `reg.getReset` but you use `adapter.getReset` below. I'm pretty sure `adapter` is the right way. Can you switch over this entire function while you're at it? (i.e. fix my mistakes.)